toppingmaker/projecttopping.py

```diff
@@ -379,11 +382,80 @@ def make_items(
 
                 self[name] = maptheme_item
 
+    class Variables(dict):
+        """
+        A dict object of dict items describing a variable according to the variable keys listed in the ExportSettings passed on parsing the QGIS project.
+        """
+
+        def make_items(
+            self,
+            project: QgsProject,
+            export_settings: ExportSettings,
+        ):
+            self.clear()
+            for variable_key in export_settings.variables:
+                self[variable_key] = QgsExpressionContextUtils.projectScope(
+                    project
+                ).variable(variable_key)
+
+    class Layouts(dict):
+        """
+        A dict object of dict items describing a layout with templatefile according to the layout names listed in the ExportSettings passed on parsing the QGIS project.
+        """
+
+        class LayoutItemProperties(object):
+            """
+            The properties of a layout item.
+            Currently it's only a layout  template file. Maybe in future here as well items or whatever can be defined.
+            """
+
+            def __init__(self):
+                # the layout  template file - if None then not requested
+                self.templatefile = None
+
+        def __init__(self):
+            self.temporary_toppingfile_dir = os.path.expanduser("~/.temp_topping_files")
```

Comment:
Can we use default functionality for this?
https://docs.python.org/3/library/tempfile.html

Makes sure it works properly across all operating systems / environments in "real" temporary locations that also will be cleaned up

Reply:
The current way is like we do it with `self.cache_path = os.path.expanduser("~/.ilicache")`.
But I do agree that this here we perfectly could move into the default temp, since we don't need those files after the step of creation.

Reply:
ilicache needs to be persistent between multiple processes (it's caching) and here we have temporary files for the current export (or not?)

Reply:
More or less. As long as the `ProjetTopping` object exists it makes sense to keep this files...
```py
# parsing project and store the temp files (after that we can throw the QgsProject away)
project_topping.parse_project(project, export_settings)
# writing YAML with project structure and copy the temp files to the path (according path resolver and target structure)
project_topping.generate_files(target)
```

Wouldn't it make more sense, not using a temp file but cleaning up instead on `generate_files`. Con here would be, that if the files are never generated, they are cleaned up neither... :thinking:

Reply:
I think the two things are not mutually exclusive:

- cleaning up makes sense
- using a temporary folder also makes sense (for the heavy addicts who use model baker in parallel in multiple qgis window's as the same time)